### PR TITLE
[rfc] refactor bookcfg

### PR DESCRIPTION
**Pull request by pietroppeter**
- it really did not make any sense to have it in book, should be in src.
- also it is not a config, it is the set of defaults of nimibook (so renamed accordingly)
- and I think it makes sense to create a `nimibook / prelude` that initializes a nimib document with nimibook defaults. it is something that you would include at beginning of document (similar to `include karax / prelude` but it does a bit more since it also initializes the nimib document)

since I made a few changes recently without discussing much, I keep this PR open for comments of contributors. @Clonkk, @hugogranstrom, @zetashift, let me know what you think, feel free also to rediscuss/criticize previous changes about bookcfg.

### CODE REVIEW

**Comment by Clonkk** [comment]
IMO ``bookcfg`` being in ``books`` is bad because ``books`` does not (currently) gets deployed by doing ``nimble install``.

Therefore, ``include ../../bookcfg`` will not work for someone who does  ``nimble install``

**Comment by pietroppeter** [comment]
> IMO `bookcfg` being in `books` is bad because `book` does not (currently) gets deployed by doing `nimble install`.
> 
> Therefore, `include ../../bookcfg` will not work for someone who does `nimble install`

agree, this is covered by this PR (`bookcfg`, which is `nimibook /defaults` in this PR is imported not included)

File: src/nimibook/pubs.nim [hunk 1]
```diff
@@ -1,7 +1,7 @@
```
Comment (zetashift) [inline src/nimibook/pubs.nim:1]: Nitpicks: blankline here and imho the filename is better as `publishers.nim`.

**Comment by zetashift** [comment]
Just before I went to sleep I thought of a similar situation, I'm glad you were able to implement it a lot faster than me haha.
Anyway I like this current approach, only thing I would like to see added in either the README or in `defaults.nim`/`prelude.nim` is how the setup works. E.g. in `prelude.nim`:
```nim
import nimib, nimibook / defaults
## Initializes a nimib document with nimibooks defaults and other stuff about what prelude.nim might do
nbInit
nbDoc.useNimibook
```
And in `defaults.nim` document what `useNimibook` exactly does.  

Of course if this is just a temporary thing then I think things are just fine as is.

**Comment by Clonkk** [comment]
Looks good to me

**Comment by HugoGranstrom** [comment]
Glad you managed the circular deps. This looks good to me as well :)

**Comment by pietroppeter** [comment]
thanks all! I will incorporate the suggestions of @zetashift and will merge later this evening. I was also planning to make a submission for this project in "[This Month in Nim](https://github.com/beef331/website)" for the month of May (to support the initiative and give visibility) and my idea was to mention you all as contributors in the description (whatever the state of commits, discussing, using, reporting issues is definitely contributions). I will update you when I have prepared a draft of the submission. the idea is also this weekend for me is to work on issues #10, #9 so that the repo is in a presentable state when people give a look at it.

**Comment by HugoGranstrom** [comment]
@pietroppeter sounds like a great idea! 😄 The more people who are aware of the greatness of Nimib and NimiBook the better! Would be an honor to be mentioned as a contributor 🤯 

Didn't even notice that hidden comment 🙃 Feels like we have come a long way in fixing the most essential of those features at least. I mean except from choosing the default theme + next/prev buttons + search there isn't that much in there that is specific to NimiBook (ie should be in Nimib). Except adding in some Nimib links of course :)

Next month maybe we could submit SciNim/getting-started to prolong the exposure of NimiBook even more 😮

**Comment by Clonkk** [comment]
> I mean except from choosing the default theme + next/prev buttons + search there isn't that much in there that is specific to NimiBook 

#12 Needs to be solved. First thing people will try is creating a book and currently there is no "proper" solution to do that.

> Next month maybe we could submit SciNim/getting-started to prolong the exposure of NimiBook even more open_mouth

For now getting-started is empty. We should probably aim to write some content before advertising it :laughing:

**Comment by HugoGranstrom** [comment]
> #12 Needs to be solved. First thing people will try is creating a book and currently there is no "proper" solution to do that.

For now, the template functionality would suffice? Or just cloning the repo if one would want to do it locally without involving a Github repo. But a nice executable would be optimal of course 🤔 And how are we going to migrate to NimiBook from what we currently have? An executable wouldn't have made it easier to transfer though. 

> For now getting-started is empty. We should probably aim to write some content before advertising it 😆

We have entire month to write content, that's lots of time... 🤣

File: src/nimibook/pubs.nim (hunk 1 above)
Comment (pietroppeter) [reply src/nimibook/pubs.nim]: done!

**Comment by pietroppeter** [comment]
added a `nimble srcpretty` task that runs `nimpretty --maxLineLen:160` over all nim files in src directory.

> added in either the README or in defaults.nim/prelude.nim is how the setup works.
> And in defaults.nim document what useNimibook exactly does.

@zetashift I will add those when working on #10

**Pull request merged**